### ISSUE TITLE
Running the DartPad examples on two API pages fails

### ISSUE THREAD

**Issue by sfshaza2** [flutter/flutter#117441]
Running the DartPad example on the [NavigationDrawer](https://api.flutter.dev/flutter/material/NavigationDrawer-class.html) pages fails with "Error compiling to JavaScript".

Running the DartPad example on the [SegmentedButton](https://api.flutter.dev/flutter/material/SegmentedButton-class.html) API page also fails with the same error.

I *am* able to run the DartPad example on the NavigationBar page, for example.

I'm on an M1 Mac laptop running Ventura 13.1. 

@HansMuller

**Comment by HansMuller** [comment]
Both of these widgets depend on relatively recent changes; are you running them against something close to the master branch?

CC @darrenaustin @QuncCccccc

**Comment by darshankawar** [comment]
Simply running https://api.flutter.dev/flutter/material/NavigationDrawer-class.html#material.NavigationDrawer.1 as-is shows the error `Error compiling to JavaScript`. The console gives below error:

```
Error compiling to JavaScript:
/tmp/dartpadBOCRTT/lib/main.dart:131:18: Error: The method 'NavigationDrawer' isn't defined for the class '_NavigationDrawerExampleState'.
 - '_NavigationDrawerExampleState' is from 'package:dartpad_sample/main.dart' ('/tmp/dartpadBOCRTT/lib/main.dart').
Try correcting the name to the name of an existing method, or defining a method named 'NavigationDrawer'.
      endDrawer: NavigationDrawer(
                 ^^^^^^^^^^^^^^^^
/tmp/dartpadBOCRTT/lib/main.dart:143:20: Error: The method 'NavigationDrawerDestination' isn't defined for the class '_NavigationDrawerExampleState'.
 - '_NavigationDrawerExampleState' is from 'package:dartpad_sample/main.dart' ('/tmp/dartpadBOCRTT/lib/main.dart').
Try correcting the name to the name of an existing method, or defining a method named 'NavigationDrawerDestination'.
            return NavigationDrawerDestination(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Also, tried the same running on dartpad.dev which is on latest stable version `Based on Flutter 3.3.10 Dart SDK 2.18.6` and throws same error.

**Comment by bruno16teixeira** [comment]
Not only I'm having this problem running the DartPad, but I also can't seem to use the Widget on my projects. I've updated everything to the latest versions, but it seems the Widget doesn't exist. 
![image](https://user-images.githubusercontent.com/7326853/212776371-7c1b6e9c-3340-4bee-a35e-256c3fd4cdcf.png)

I've checked the SDK installed on my machine it seems there's no SegmentedButton or NavigationDrawer in files such as `material.dart` or `theme_data.dart` even though I have v3.3.10
![image](https://user-images.githubusercontent.com/7326853/212776510-f87af235-37e7-4320-8f73-abce6126abe7.png)

I've downloaded the source code from this release: https://github.com/flutter/flutter/releases/tag/3.3.10 and checked those files again, and nothing there. 

This PR was merged into master before 3.3.10 release but it seems it was not included in the source code uploaded.
https://github.com/flutter/flutter/pull/113723

Hopefully, this sheds some light on what might be going on.
